source/common/router/router.cc

```diff
@@ -637,23 +726,25 @@ bool Filter::maybeRetryReset(Http::StreamResetReason reset_reason) {
   return false;
 }
 
-void Filter::onUpstreamReset(Http::StreamResetReason reset_reason) {
-  ASSERT(upstream_request_);
+void Filter::onUpstreamReset(Http::StreamResetReason reset_reason,
+                             UpstreamRequest* upstream_request) {
```

Comment:
Let me know if it's weird to be using raw pointers here, alternatively could make UpstreamRequest a shared pointer (it's a unique pointer)

Reply:
You can probably pass a ref instead since it won't ever be null